src/test/java/FakeUI.java

```diff
@@ -0,0 +1,33 @@
+import java.util.ArrayList;
+
+class FakeUI extends UI {
```

Comment:
It might get pretty confusing that there are multiple (well, at least 2) classes named `FakeUI` - probably want to standardize on one. I didn't look for additional duplicates, but that's probably a good idea to do while you're thinking about it.